invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/Identifiers/PIDField.js

```diff
@@ -349,11 +377,16 @@ class CustomPIDField extends Component {
   constructor(props) {
     super(props);
 
-    const { canBeManaged, canBeUnmanaged } = this.props;
+    const { canBeManaged, canBeUnmanaged, record, field } = this.props;
     this.canBeManagedAndUnmanaged = canBeManaged && canBeUnmanaged;
-
+    const value = field?.value;
+    const isDraft = record?.is_draft;
     this.state = {
-      isManagedSelected: undefined,
+      isManagedSelected:
+        isDraft === true && value?.identifier && value?.provider !== PROVIDER_EXTERNAL
+          ? true
+          : undefined,
+      isNoNeedSelected: undefined,
```

Comment:
```suggestion
   const isInternalProvider = value?.provider !== PROVIDER_EXTERNAL;
   const isDraft = record?.is_draft == true;
   const hasIdentifier = value?.identifier;
   const isManagedSelected =  isDraft && hasIdentifier && isInternalProvider ? true : undefined;
  
   this.state = {
      isManagedSelected: isManagedSelected;
      isNoNeedSelected: undefined,
```

also, why do we set isManagedSelected to `undefined` not `false`?